src/ripple/app/ledger/Ledger.test.cpp

```diff
@@ -23,292 +23,493 @@
-        tx_json["Flags"] = tfClearNoRipple;
+        tx_json["LimitAmount"] = Amount{amountCurrency, trusted}.getJson();
         STTx tx = parseTransaction(from, tx_json);
-        applyTransaction(ledger, tx);
+        return applyTransaction(tx);
     }
 
-    Ledger::pointer
-    close_and_advance(Ledger::pointer ledger, Ledger::pointer LCL)
+    SLE::pointer
+    getSLEi(uint256 const& index)
+    {
+        return ledger->getSLEi(index);
+    }
+
+    SLE::pointer
+    getSLE(uint256 const& index)
     {
-        SHAMap::pointer set = ledger->peekTransactionMap();
-        CanonicalTXSet retriableTransactions{set->getHash()};
-        Ledger::pointer newLCL = std::make_shared<Ledger>(false, *LCL);
-        // Set up to write SHAMap changes to our database,
-        //   perform updates, extract changes
-        applyTransactions(set, newLCL, newLCL, retriableTransactions, false);
-        newLCL->updateSkipList();
-        newLCL->setClosed();
-        int asf = newLCL->peekAccountStateMap()->flushDirty(hotACCOUNT_NODE,
-                                                        newLCL->getLedgerSeq());
-        int tmf = newLCL->peekTransactionMap()->flushDirty(hotTRANSACTION_NODE,
-                                                        newLCL->getLedgerSeq());
-        using namespace std::chrono;
-        auto const epoch_offset = days{10957};  // 2000-01-01
-        std::uint32_t closeTime = time_point_cast<seconds>  // now
-                                         (system_clock::now()-epoch_offset).
-                                         time_since_epoch().count();
-        int CloseResolution = seconds{LEDGER_TIME_ACCURACY}.count();
-        bool closeTimeCorrect = true;
-        newLCL->setAccepted(closeTime, CloseResolution, closeTimeCorrect);
-        return newLCL;
+        return ledger->getSLE(index);
     }
 
-    void test_genesisLedger ()
+    std::uint32_t
+    ledger_accept()
     {
-        std::uint64_t const xrp = std::mega::num;
+        close_and_advance_by_ref(ledger, LCL);
+        return ledger->getLedgerSeq();
+    };
 
-        // Create master account
-        auto master = createAccount();
+    /* Something about something */
+    TestAccount rootAccount;
+    Ledger::pointer LCL;
+    Ledger::pointer ledger;
 
+    TestContext () :
+        rootAccount(createAccount("masterpassphrase")),
         // Create genesis ledger
-        Ledger::pointer LCL = createGenesisLedger(100000*xrp, master);
-
+        LCL(createGenesisLedger(100000000000000000ull, rootAccount)),
         // Create open scratch ledger
-        Ledger::pointer ledger = std::make_shared<Ledger>(false, *LCL);
+        ledger(std::make_shared<Ledger>(false, *LCL)) {}
+};
+
+class Ledger_test : public TestSuite
+{
+    using TestAccount = TestContext::TestAccount;
+    using TXResult = TestContext::TXResult;
+
+    void
+    test_TestContext ()
+    {
+        #define TEST_ACCOUNT(name) auto name = c.createAccount(#name);
```

Comment:
Well when you put it that way it sounds much better but the macros are a bitter pill